arwen/host/asyncCall.go

```diff
@@ -298,6 +313,17 @@ func (host *vmHost) createCallbackContractCallInput(
 		Function:      callbackFunction,
 	}
 
+	if isESDTOnCallBack {
+		contractCallInput.Function = core.BuiltInFunctionESDTTransfer
+		contractCallInput.Arguments = make([][]byte, 0, len(arguments))
+		contractCallInput.Arguments = append(contractCallInput.Arguments, destinationVMOutput.ReturnData[1], destinationVMOutput.ReturnData[2])
```

Comment:
Suggest to extract `destinationVMOutput.ReturnData[1]` and `destinationVMOutput.ReturnData[2]` into variables such as `esdtTokenName` and `esdtValue`.